ykrt/src/compile/jitc_yk/codegen/x64/mod.rs

```diff
@@ -1659,22 +1902,55 @@ pub(super) struct X64CompiledTrace {
     gdb_ctx: GdbCtx,
 }
 
+impl X64CompiledTrace {
+    /// Return the locations of the live variables this trace expects as input.
+    fn entry_vars(&self) -> &[VarLocation] {
+        &self.entry_vars
+    }
+}
+
 impl CompiledTrace for X64CompiledTrace {
     fn entry(&self) -> *const libc::c_void {
         self.buf.ptr(AssemblyOffset(0)) as *const libc::c_void
     }
 
     fn sidetraceinfo(&self, gidx: GuardIdx) -> Arc<dyn SideTraceInfo> {
-        // FIXME: Can we reference these instead of copying them?
-        let aotlives = self.deoptinfo[&usize::from(gidx)]
+        // FIXME: Can we reference these instead of copying them, e.g. by passing in a reference to
+        // the `CompiledTrace` and `gidx` or better a reference to `DeoptInfo`?
+        let deoptinfo = &self.deoptinfo[&usize::from(gidx)];
+        let lives = deoptinfo
             .live_vars
             .iter()
-            .map(|(iid, _)| iid.clone())
+            .map(|(a, l)| (a.clone(), l.into()))
             .collect();
-        let callframes = self.deoptinfo[&usize::from(gidx)].inlined_frames.clone();
+        let callframes = deoptinfo.inlined_frames.clone();
+
+        // Get the root trace from the HotLocation
```

Comment:
I think this should have `FIXME: It might be better if we had a consistent mechanism for doing this rather than fishing it out of the `HotLocationKind` or similar.

Reply:
[1e69032](https://github.com/ykjit/yk/pull/1422/commits/1e69032bcb639fd3c88fc00fab48075378743169)